pkg/operator/operator_test.go

```diff
@@ -482,10 +521,42 @@ func TestSync(t *testing.T) {
 					withDaemonSetGeneration(3, 1),
 					withDaemonSetStatus(replica1, replica1, replica1)),
 				ebsCSIDriver: ebsCSIDriver(
-					withStatus(replica2), // 1 deployment + 1 daemonSet
-					withGenerations(3),   // now the operator knows generation 1
+					withStatus(replica2),     // 1 deployment + 1 daemonSet
+					withGenerations(3, 3, 1), // now the operator knows generation 1
 					withTrueConditions(opv1.OperatorStatusTypeAvailable, opv1.OperatorStatusTypeUpgradeable, opv1.OperatorStatusTypePrereqsSatisfied, opv1.OperatorStatusTypeProgressing), // Progressing due to Generation change
 					withFalseConditions(opv1.OperatorStatusTypeDegraded)),
+				credentialsRequest: getCredentialsRequest(withCredentialsRequestGeneration(1)),
+			},
+		},
+		{
+			// CredentialRequests is modified by user, and gets replaced by the operator.
+			name:   "CredentialRequests modified by user",
+			images: defaultImages(),
+			initialObjects: testObjects{
+				deployment: getDeployment(argsLevel2, defaultImages(),
+					withDeploymentReplicas(1),
+					withDeploymentGeneration(1, 1),
+					withDeploymentStatus(replica1, replica1, replica1)),
+				daemonSet: getDaemonSet(argsLevel2, defaultImages(),
+					withDaemonSetGeneration(1, 1),
+					withDaemonSetStatus(replica1, replica1, replica1)),
+				ebsCSIDriver:       ebsCSIDriver(withGenerations(1, 1, 1)),                     // the operator knows the old generation of the Deployment
+				credentialsRequest: getCredentialsRequest(withCredentialsRequestGeneration(2)), // modified by user
+			},
+			expectedObjects: testObjects{
+				deployment: getDeployment(argsLevel2, defaultImages(),
+					withDeploymentReplicas(1),
+					withDeploymentGeneration(1, 1),
+					withDeploymentStatus(replica1, replica1, replica1)),
+				daemonSet: getDaemonSet(argsLevel2, defaultImages(),
+					withDaemonSetGeneration(1, 1),
+					withDaemonSetStatus(replica1, replica1, replica1)),
+				ebsCSIDriver: ebsCSIDriver(
+					withStatus(replica2),     // 1 deployment + 1 daemonSet
+					withGenerations(1, 1, 3), // now the operator knows generation 3
+					withTrueConditions(opv1.OperatorStatusTypeAvailable, opv1.OperatorStatusTypeUpgradeable, opv1.OperatorStatusTypePrereqsSatisfied),
+					withFalseConditions(opv1.OperatorStatusTypeDegraded, opv1.OperatorStatusTypeProgressing)),
+				credentialsRequest: getCredentialsRequest(withCredentialsRequestGeneration(1)),
```

Comment:
minor nit: should we actually verify if operator indeed restores spec field rather than just verifying generation?

Reply:
Added check for expected / actual `credentialsRequest`.